micro-rdk/src/common/registry.rs

```diff
@@ -352,20 +370,20 @@ impl ComponentRegistry {
 
     pub(crate) fn get_dependency_function(
         &self,
-        component_type: &'static str,
-        model_name: &str,
+        component_type: String,
```

Comment:
If these are just used for lookups, could they remain `&str`? Here and elsewhere for lookups.

Reply:
specifically `component_type` right? or did you mean `component_type` and `model_name`?

Reply:
Both, I think, if it works.